libcxx/test/std/thread/thread.mutex/thread.lock/thread.lock.guard/assign.compile.fail.cpp

```diff
@@ -6,6 +6,8 @@
 //
 //===----------------------------------------------------------------------===//
 
+// UNSUPPORTED: no-threads
```

Comment:
We can lift this `UNSUPPORTED` by defining our own mutex-like class. Also, while we're at it this test should be a `.compile.pass.cpp` and it should test `static_assert(!std::is_copy_assignable<std::lock_guard<MyMutexType>>::value, "");`.

Reply:
Done.